### DISCUSSION TITLE
ESM Support in Scripts

### DISCUSSION THREAD

**Discussion by RzGabriel** [usebruno/bruno#3535] [Q&A]
First and foremost, thanks for this amazing project! Testing my API endpoints became much easier with Bruno, I'm recommending it to everyone I can.

After playing around a little bit with the scripting module in Bruno, I noticed that only CommonJS syntax is supported in scripts. This means that external modules that are written in the newer ECMAScript module syntax cannot be imported.

Are there any plans for adding ESM support?

**Comment by centur** [comment]
Hi, I have the same question, more and more modules are getting ESM only in latest versions: commonly used in my scripts is `faker-js/faker` - from version 10 - it's esm only, so  with ^10.0.0 usual `const {faker} = require('@faker-js/faker')` is no longer an option 
Bruno produces `'import' and 'export' may only appear at the top level`
cc  @helloanoop  or any other devs - any chance this can be considered ?

**Reply by helloanoop** [reply]
@centur Onboard to support ESM scripts and have added to our backlog. 

As of today Bruno supports two modes
* Safe Mode - uses quickjs runtime
* Developer Mode - uses vm2 runtimr

The vm2 runtime is being deprecated and we will move to native node runtime in v3.

When we build ESM support, it needs to work in quickjs as well as Node VM.
